rules/why-upgrade-to-latest-dotnet/rule.md

```diff
@@ -0,0 +1,49 @@
+---
+type: rule
+archivedreason: 
+title: Do you know why to upgrade from .NET Framework to the latest .NET?
+guid: 75e296c8-1c21-4a4e-a2fb-4df32fd6140e
+uri: why-upgrade-to-latest-dotnet
+created: 2023-10-06T23:18:38.515Z
+authors:
+  - title: Yazhi Chen
+    url: https://www.ssw.com.au/people/yazhi-chen
+
+---
+In the world of software development, staying up-to-date with technology is not just a trend; it's a necessity. As the digital landscape evolves, so do the tools and frameworks that developers rely on to build applications. One significant transition that has been taking place in recent years is the migration from the legacy .NET Framework to the latest .NET (currently .NET 7 as of November 2022 and .NET 8 will be released on November 2023) offerings. But why is this shift so crucial, and what benefits does it bring to the table?
+<!--endintro-->
+
+## The Evolution of .NET
+To understand the significance of upgrading from .NET Framework to the latest .NET, let's first take a brief look at the evolution of the .NET platform.
+
+### .NET Framework: The Legacy
+
+The .NET Framework has been the backbone of Windows application development for nearly two decades. It has provided a robust and versatile environment for building Windows desktop applications, web applications, and services. However, as technology progressed, the limitations of .NET Framework became more apparent.
+
+#### The Pain Point:
+
+* Compatibility: .NET Framework applications primarily run on Windows, limiting cross-platform deployment.
+* Performance: Older versions of .NET Framework can be resource-intensive and less efficient.
+* Security: The framework reached a point where it was no longer actively developed, leaving applications vulnerable to security risks.
+* Modern development features: .NET Framework may not support some of the modern development features and patterns that are essential for staying competitive in today's tech landscape.
```

Comment:
Phrase positively?

```suggestion
* Modern development features: Newer versions of .NET give us more opportunities to design products and services using many newer capabilities that are not backported to .NET Framework. This allows us to build, test and deploy faster, and more securely than ever before, keeping us competitive in today's tech landscape.
```

You might want to double-check the line to make sure I've not gone too far to the other side 😅

Reply:
This is the same, in this section, I want to focus on the pain point of the .NET Framework and why we do not want to stay in .NET Framework anymore.

Reply:
It's just a suggestion 🙂. You can illustrate pain points through a positive for the alternative. 

Saying things like may not support modern development features makes it sound like you don't know if it does or not, I'd then make that be more assertive, as in, it doesn't support modern development features and patterns. 
Although the sentence is not completely accurate, as it depends on the business use case. 

This is where showing the alternative as the pain point comes across better

Reply:
Updated